students/jstonham/README.md

```diff
@@ -2618,23 +2618,66 @@ Indices are generally stored in RAM, which is partly why they are so performant.
 - Redis - stored in RAM so does not require indices as it's already very performant.
 - ElasticSearch - useful for searching. As programmers generally use it to speed up searching, they tend to only store the search-related data here.
 
-## Creating Websites
+## Important Tools for Creating Websites
 
-React is a JavaScript library for building user interfaces. It takes the backend state and turns it into a frontend view.
+### React
 
-Gatsby JavaScript Framework is a tool based on React that builds websites using only static files, which increases the speed the user can navigate through the resulting interface.
+React is a JavaScript library or framework for building user interfaces.
+It takes the backend state and turns it into a frontend view.
+The framework takes care of the boilerplate so React allows programmers to simplify tasks that would otherwise be difficult.
+(In a similar way, JUnit is a Java framework for simplifying the writing of tests).
 
-Node.js provides the ability for programmers to write web servers using JavaScript. Node Package Manager (NPM) is a command line interface for managing node modules (a way of getting node applications such as VS Code and Atom).
+There is some debate regarding whether React should be thought of as a framework or a library.
+Due to React's popularity, many tools have been built around it, e.g. JSX (JavaScript Extended).
+React, in combination with these tools, is generally thought of as a framework.
+
+React is a powerful tool that allows classes, inheritance, nested components etc.
+React also allows JavaScript, HTML and CSS to be combined together in website components.
+
+### Gatsby
+
+Gatsby JavaScript Framework is a tool that builds websites using only static files, which increases the speed the user can navigate through the resulting interface. It is good for front-end development.
+
+Gatsby is designed to run in a web browser.
+It is a combination of React and other libraries, with the `gatsby-cli` (command line tool) in `Node.js`.
+
+### Node
+
+Node.js provides the ability for programmers to write web servers using JavaScript.
```

Comment:
We should distinguish here between "writing applications that are web servers" (although Node has support for this), and writing code that can be run on a server platform (basically in a shell). It's the latter that Node permits. This might seem odd, as basically any of the popular programming languages you know about supports and might even require this platform. E.g.:

- Java
- C
- C#
- C++
- Go
- Ruby
- Python

All of these were written so that, if you knew the right incantation, you could run it in a terminal.

**JavaScript** on the other hand was written to run in the browser. It's only with the advent of Node.js that we can also run it outside the browser (and in particular in a shell).

```suggestion
Node.js provides the ability for programmers to write JavaScript on servers.
```